infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -340,11 +340,10 @@ private void connectToClusterManager() {
      * - Start Log Replication Server(listens and processes incoming requests from the Source)
      * Both:
      * - Metadata Managers which maintain metadata related to replication and its status
-     * @param role
      */
-    private void performRoleBasedSetup(ClusterRole role) {
-        if (role != ClusterRole.SOURCE && role != ClusterRole.SINK) {
-            log.debug("Cluster role is {}.  Not performing role-based setup.", role);
+    private void performRoleBasedSetup() {
+        if (topologyDescriptor.getRemoteSinkClusters().isEmpty() && topologyDescriptor.getRemoteSourceClusters().isEmpty()) {
```

Comment:
do we need to handle the case that the set is null?

Reply:
No, its never going to be null.
The topologyDescriptor's constructor initializes them to an empty map